server/src/main/java/io/crate/execution/engine/indexing/BatchIteratorBackpressureExecutor.java

```diff
@@ -155,7 +159,7 @@ private void consumeIterator() {
             return;
         }
         try {
-            while (batchIterator.moveNext()) {
+            while (!consumptionFinished && batchIterator.moveNext()) {
```

Comment:
Doesn't this have to take into account whether `failFast` is set? Otherwise this would generally stop processing records once an error occurs.

Reply:
if `failFast` is set, the `UpsertResultCollector` implementation is wrapped with `FailFastResultCollector` which would throw and eventually set `consumptionFinished`.